pkg/client/client.go

```diff
@@ -0,0 +1,209 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/openshift/cluster-network-operator/pkg/util/k8s"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	kinformer "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	osoperclient "github.com/openshift/client-go/operator/clientset/versioned"
+	osoperinformer "github.com/openshift/client-go/operator/informers/externalversions"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operv1 "github.com/openshift/api/operator/v1"
+	netopv1 "github.com/openshift/cluster-network-operator/pkg/apis/network/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const defaultResyncPeriod = 5 * time.Minute
+
+// Client is a bag of holding for object clients & informers.
+// It is generally responsible for managing informer lifecycle.
+//
+// int 5, wis 2, dex 10, cha 6
+type Client struct {
+	cfg *rest.Config
+
+	// Same configuration, but with protobuf enabled
+	// Only to be used for proper k8s api types
+	protocfg *rest.Config
+
+	kClient  kubernetes.Interface
+	kFactory kinformer.SharedInformerFactory
+
+	// client for config.openshift.io
+	operClient  *osoperclient.Clientset
+	operFactory osoperinformer.SharedInformerFactory
+
+	// restMapper is the mapper from GVK to GVR (among other fun tasks)
+	restMapper meta.RESTMapper
+	// dynamic is an untyped, uncached client for making direct requests
+	// against the apiserver.
+	dynamic crclient.Client
+
+	// informers is any other Informer we create, e.g. ones with
+	// specific watches, that are't managed by the factories.
+	informers []cache.SharedInformer
+
+	// OperatorClient is used by the library-go helpers
+	oc *OperatorClient
```

Comment:
The name `oc` is confusing because `oc` means something else.
Likewise `OperatorClient` vs `operClient`...

Reply:
yeah, lemme think up a new naming scheme.

Reply:
okay, tightened up the naming. Now the type is called `OperatorHelperClient` which stinks but it's basically a private type anyways.